src/main/java/Scorers/Implementations/ScorerImpl.java

```diff
@@ -2,51 +2,38 @@
 
 import Entities.Ingredient;
 import Entities.Recipe;
+import Entities.RecipeItem;
 import Entities.Tag;
 import Matchers.Implementations.IngredientMatcher;
 import Matchers.Implementations.TagMatcher;
 import Scorers.Scorer;
 import Storages.RecipeStorage;
 
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
-import java.util.Collections;
-import java.util.Comparator;
-import java.lang.Double;
+import java.util.Set;
 
-public class ScorerImpl implements Scorer {
 
-    final List<Ingredient> ingredients;
-    final List<Tag> tags;
-    final String name;
-    Recipe recipe;
+public class ScorerImpl implements Scorer {
 
     final IngredientMatcher ingredientMatcher;
-    final TagMatcher tagMatcher;
 
-    public ScorerImpl(List<Ingredient> ingredients, List<Tag> tags, String name) {
-        this.ingredients = ingredients;
-        this.tags = tags;
-        this.name = name;
-        this.ingredientMatcher = new IngredientMatcher(ingredients);
-        this.tagMatcher = new TagMatcher(tags);
+    public ScorerImpl(List<Recipe> recipes) {
+        List<Ingredient> ingredientList = new ArrayList<>();
+        for (Recipe recipe : recipes)
+            for (RecipeItem recipeItem : recipe.items())
+                ingredientList.add(recipeItem.ingredient());
+        ingredientList = new ArrayList<> (new HashSet<>(ingredientList));
+        this.ingredientMatcher = new IngredientMatcher(ingredientList);
     }
 
-    public double score() {
-        //private final double NAME_VAL = 0.7; to be used in future updates
-        double INGREDIENT_VAL = 0.7;
-        double ingredientScore = INGREDIENT_VAL * this.ingredientMatcher.floatMatch(this.recipe);
-        double TAG_VAL = 0.3;
-        double tagScore = TAG_VAL * this.tagMatcher.floatMatch(this.recipe);
-        return ingredientScore + tagScore;
+    @Override
+    public double score() { //candidate for removal
+        return 0;
     }
 
-    /**
-     * Take in a recipe storage and return a list of the top 10 (less if there isn't 10) most similar recipes
-     * @param recipes list of recipes
-     * @param num number of recipes to return
-     * precondition: num <= recipes.size()
-     */
+    @Override
     public List<Recipe> returnNumRecipes(RecipeStorage recipes, int num) {
         num = Math.min(num, recipes.size());
```

Comment:
Good idea though for generalizing this, the concrete number 10 was not good practice. We need it to work for x amount of recipes.

Reply:
Don't name things like this, variables or functions. `num` is meaningless. The naming of this function is also poor. We know that the function returns something because of the return type, so we never need to start function names with `return`. So, `numRecipes`, well now this means "how many recipes do you have" which is not what the function does. Based on the return type, it returns recipes so maybe we just call it "recipes", but it actually scores our recipes (from the sort call) so really this scores recipes and returns the top recipes. So turns out that this is actually the `score` function. So, get rid of this score function (and on the interface), rename this function to `scoreRecipes` or something (and on the interface), and we don't need any num param, change this to `limit` which is more conventional to denote that it limits the number of results.

Also, what happens if the num is 0 or less, well you get a big old error.

Reply:
Duly noted. I'll change it so that **returnNumRecipes** becomes **score**, and I'll put that in the Scorer.